crates/re_space_view_spatial/src/parts/assets.rs

```diff
@@ -29,51 +29,92 @@ impl Default for Asset3DPart {
-
-            let mesh = ctx.cache.entry(|c: &mut MeshCache| {
-                c.entry(
-                    &ent_path.to_string(),
-                    picking_instance_hash.versioned(primary_row_id),
-                    AnyMesh::Asset(&mesh),
-                    ctx.render_ctx,
-                )
-            });
-            if let Some(mesh) = mesh {
-                instances.extend(mesh.mesh_instances.iter().map(move |mesh_instance| {
-                    MeshInstance {
+        let transform = arch_view
+            .iter_raw_optional_component::<OutOfTreeTransform3D>()?
+            .and_then(|mut batch| batch.next());
+
+        let mesh = Asset3D {
+            data: arch_view
+                .iter_required_component::<Blob>()?
+                .next()
+                .ok_or_else(|| re_types::DeserializationError::MissingData {
+                    backtrace: re_types::_Backtrace::new_unresolved(),
+                })?,
+            media_type: arch_view
+                .iter_raw_optional_component::<MediaType>()?
+                .and_then(|mut batch| batch.next()),
+            // NOTE: Don't even try to cache the transform!
+            transform: None,
+        };
+
+        let primary_row_id = arch_view.primary_row_id();
+        let picking_instance_hash = re_data_store::InstancePathHash::entity_splat(ent_path);
+        let outline_mask_ids = ent_context.highlight.index_outline_mask(InstanceKey::SPLAT);
+
+        // TODO(#3232): this is subtly wrong, the key should actually be a hash of everything that got
+        // cached, which includes the media type...
+        let mesh = ctx.cache.entry(|c: &mut MeshCache| {
+            c.entry(
+                &ent_path.to_string(),
+                picking_instance_hash.versioned(primary_row_id),
+                AnyMesh::Asset(&mesh),
+                ctx.render_ctx,
+            )
+        });
+
+        if let Some(mesh) = mesh {
+            let mut mesh_instances = mesh
+                .mesh_instances
+                .iter()
+                .map(move |mesh_instance| MeshInstance {
+                    gpu_mesh: mesh_instance.gpu_mesh.clone(),
+                    world_from_mesh: ent_context.world_from_obj * mesh_instance.world_from_mesh,
+                    outline_mask_ids,
+                    picking_layer_id: picking_layer_id_from_instance_path_hash(
+                        picking_instance_hash,
+                    ),
+                    ..Default::default()
+                })
+                .collect_vec();
+
+            // Apply the out-of-tree transform.
+            if let Some(transform) = transform.as_ref() {
+                let (scale, rotation, translation) =
+                    glam::Affine3A::from(transform.0).to_scale_rotation_translation();
+                let transform =
+                    glam::Affine3A::from_scale_rotation_translation(scale, rotation, translation);
```

Comment:
this extra dance should be justified. Why are we throwing away shearing?

Reply:
That's copy pasted as-is from the original code... I don't know why we're throwing the shear :man_shrugging:

Reply:
Huh strange! To me this looks like from before we allowed general affine transforms - for a quite a while we only had rigid + uniform scale, but nowadays we do the correct thing to the normals (https://github.com/rerun-io/rerun/blob/main/crates/re_renderer/src/renderer/mesh_renderer.rs#L228).
Away with it! :)